aws-xray-recorder-sdk-apache-http/src/main/java/com/amazonaws/xray/proxies/apache/http/TracedHttpClient.java

```diff
@@ -104,7 +104,16 @@ public static void addRequestInformation(Subsegment subsegment, HttpRequest requ
         Segment parentSegment = subsegment.getParentSegment();
 
         if (subsegment.shouldPropagate()) {
-            request.addHeader(TraceHeader.HEADER_KEY, TraceHeader.fromEntity(subsegment).toString());
+            // If the trace is not sampled, passing Parent and Sampled won't matter
+            TraceHeader t = TraceHeader.fromEntity(subsegment);
+            if (t.getSampled() != TraceHeader.SampleDecision.SAMPLED) {
+                request.addHeader(
+                        TraceHeader.HEADER_KEY,
+                        "Root=" + t.getRootTraceId().toString());
```

Comment:
I am a bit confused with this line, does it drop Parent and Sampled? I.e. if we have header `Root=...;Parent=...;Sampled=0` will we propagate only `Root=...`?
If so, it can cause undesired behaviour if the downstream in active tracing mode because `Root=...`-only header can cause sampling decision to be `1`. I.e. if the upstream decides not to sample then the downstream with active tracing can change the decision.

Reply:
Changed logic to detect passthrough mode via `NoOpSubSegment` instead of relying on sampling status.

This is because `SamplingDecision.UNKNOWN` is never the outcome of getting the trace header from the subsegment directly, so we need to rely on the type of the subsegment.